src/Services/Qrda/Helpers/View.php

```diff
@@ -21,11 +21,17 @@ trait View
 
     public function measures()
     {
-        // TODO: we don't know if this is the correct implementation or not, depends on how the measures are sent.
-        $hqmf_id = $this->_measures['hqmf_id'] ?? null;
-        $hqmf_set_id = $this->_measures['hqmf_set_id'] ?? null;
-        $description = $this->_measures['description'] ?? null;
-        return json_encode(['hqmf_id' => $hqmf_id, 'hqmf_set_id' => $hqmf_set_id, 'description' => $description]);
+        // Limit the pollution of the context by only passing the values we need
+        return array_map(
+            function ($measure) {
+                return [
+                    'hqmf_id' => $measure['hqmf_id'] ?? null,
+                    'hqmf_set_id' => $measure['hqmf_set_id'] ?? null,
+                    'description' => $measure['description'] ?? null
+                ];
+            },
+            $this->_measures
+        );
```

Comment:
I assume this will filter what QDMs get reported in patient section. If not, we need to filter populating the patient section with only data appliable to the measure(s) being reported.

Reply:
No that's not how it works. The QDMs for patient are all populated (diagnosis, observations, allergies, etc), then you run the measures based on the data. At that point we don't even really know what data we'll need top calculate the measures. 

You don't then remove patient data that wasn't needed to calculate the measures.

The measure section contains one or more measures that are being reported, and lists their Description and ID, and the results are presumably contained in linked "external documents" which may be Cat iii, I'm not sure, may need to get clarification from CMS.

Reply:
I'm pretty sure. The patient data section should only be populated with the data relative to the measure being reported. Not all possible qdms.
My understanding is that the qdms are resolved based on the measures QDEs and are generally only positive results.

```
eMeasure Data Elements
• eMeasures are comprised of the data elements that are
required to compute them
• Identify QDEs in the referenced eMeasure(s)
• Example: Medication Administered - Aspirin
Find matching data elements in a patient chart
• Specification calls this “scooping”
• Example: Find all medications administered for a patient
Filter the data to the relevant elements
• Example: Only place administrations of Aspirin in the
QRDA- I document
• This happens by matching clinical codes
• Specification calls this “smoking gun”
Repeat for all QDEs in the referenced
eMeasure(s)
```

I don't know what would happen in testing the way we have now though, I'm with you, why not send all.

Perhaps we need a phone call to clarify with each other...

Reply:
From a Lantana doc for MU2 available at https://ecqi.healthit.gov/system/files/qrda_basics_08_12_2015_a_508.pdf.
It just gives an outline on QRDA.
I'm looking over the IG now.

Reply:
@kchapple If  I read this correctly then appears you may be correct. From HL7 IG
```
6.2.3 How Much Data Should be Sent?
QDM-based QRDA adheres to a "scoop and filter" philosophy, whereby data from an 
EHR are scooped up and filtered, and the remaining content is packaged into the 
instance.
When the recipient of the instance has access to no other EHR data, it is important that 
the instance include data elements relevant to computing eCQM criteria, as well as the 
other data elements defined in an eCQM —for stratification, for risk adjustment, etc. 
Every data element present in the EHR that is required by the referenced eCQM(s), not 
just those needed to compute criteria, shall be included in the QRDA document.
The EHR may have more data than are relevant to the referenced eCQM(s) and more 
data than are needed to compute the criteria. For instance, a patient who has been in 
the Intensive Care Unit undergoing continuous blood pressure monitoring will have 
reams of blood pressure observations. QDM-based QRDA adheres to a "smoking gun"
philosophy where, at a minimum, the conclusive evidence needed to confirm that a 
criterion was met shall be included in the instance.
At the very least, the QRDA document should include:
• For each data element in each referenced eCQM, smoking gun data that offer 
confirmatory proof, where a patient has met the criterion—For disjunctive criteria
Page 45 HL7 CDA R2:QRDA I, Release 1, STU Release 5 US Realm, Vol. 1 — Introductory Material
© 2017 Health Level Seven, International. All rights reserved. December 2017
(i.e., where a criterion can be satisfied by one of multiple data elements) include 
minimal smoking gun data for at least one data element.
• Stratification variables, supplemental data elements, risk adjustment variables, and 
any other data element specified in the referenced eCQM(s)
A quality program implementing QRDA will often provide prescriptive guidelines that 
define additional data, outside the smoking gun, that may or must be sent (such as the 
complete problem or medication list). Where such prescriptive guidelines exist, those 
take precedence over the more general guidance provided here. In other words, the 
“smoking gun” heuristic ensures that the minimum is present in the QRDA, and does 
not preclude inclusion of additional data.
```

Reply:
This is my takeaway:

> 
The EHR may have more data than are relevant to the referenced eCQM(s) and more data than are needed to compute the criteria.

Reply:
To me, it sounds like they need "at least" the relevant data, which they call the "smoking gun", not ONLY the relevant data.

Reply:
agreed Ken. going by Cypress sample docs they only include 'smoking gun' which makes it easier to eval the measure visually. Our html version of the doc will look very similar across patients and measures however, if ONC/CMS/HL7 doesn't care then I suppose we shouldn't either.

Still need our organizational/author/addresses/phone etc in our context.
CCM has all these queries if want to reuse them. Unsure how you want to populate in patient data class/context.

I'm working on Cat I export.